File: data/readable_format/cpp/file2411.cpp
#include <iostream>
#include <string>
#include <cmath>

// Define the function prototypes for gsw_saar and gsw_error_code
double gsw_saar(double p, double lon, double lat);
double gsw_error_code(int code, const std::string &func_name);

// Define constants for error limits
const double gsw_error_limit = /* put appropriate value here */;

// Function gsw_fdelta
double gsw_fdelta(double p, double lon, double lat) {
    const std::string func_name = "gsw_fdelta";
    double saar = gsw_saar(p, lon, lat);
    
    if (saar > gsw_error_limit) {
        return gsw_error_code(1, func_name);
    } else {
        return ((1.0 + 0.35) * saar) / (1.0 - 0.35 * saar);
    }
}

// Dummy implementations for gsw_saar and gsw_error_code
double gsw_saar(double p, double lon, double lat) {
    // Placeholder implementation
    // Replace with actual logic or library call
    return 0.0;
}

double gsw_error_code(int code, const std::string &func_name) {
    // Placeholder implementation
    // Replace with actual logic or library call
    std::cerr << "Error in function " << func_name << ": code " << code << std::endl;
    return std::nan("");
}

int main() {
    // Example usage
    double p = 1000.0;
    double lon = 30.0;
    double lat = -60.0;
    
    double result = gsw_fdelta(p, lon, lat);
    std::cout << "gsw_fdelta: " << result << std::endl;
    
    return 0;
}

File: data/readable_format/fortran/file2411.f90
program test_gsw_fdelta
    use, intrinsic :: iso_fortran_env, only: real64
    implicit none

    real(real64) :: p, long, lat, result

    ! Test values
    p = 1000.0_real64
    long = 30.0_real64
    lat = -60.0_real64

    ! Call the function and print the result
    result = gsw_fdelta(p, long, lat)
    print *, "gsw_fdelta(", p, ", ", long, ", ", lat, ") = ", result

contains

    function gsw_fdelta(p, long, lat) result(fdelta)
        real(real64), intent(in) :: p, long, lat
        real(real64) :: fdelta
        real(real64) :: saar
        real(real64), parameter :: gsw_error_limit = 0.02_real64
        character(len=*), parameter :: func_name = "gsw_fdelta"

        saar = gsw_saar(p, long, lat)

        if (saar > gsw_error_limit) then
            fdelta = gsw_error_code(1, func_name)
        else
            fdelta = ((1.0_real64 + 0.35_real64) * saar) / (1.0_real64 - 0.35_real64 * saar)
        end if
    end function gsw_fdelta

    real(real64) function gsw_saar(p, long, lat)
        real(real64), intent(in) :: p, long, lat
        gsw_saar = 0.01_real64  ! Dummy value for demonstration
    end function gsw_saar

    real(real64) function gsw_error_code(code, func_name)
        integer, intent(in) :: code
        character(len=*), intent(in) :: func_name
        print *, "Error in function ", trim(func_name), ": code ", code
        gsw_error_code = -9999.0_real64  ! Dummy error value for demonstration
    end function gsw_error_code

end program test_gsw_fdelta